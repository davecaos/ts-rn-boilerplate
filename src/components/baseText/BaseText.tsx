import React from 'react'
import {Text, TextProps} from 'react-native'

import {withTheme, IThemeProps} from '../../services/themes'

interface BaseTextProps extends IThemeProps, TextProps {
  text?: string
  fontFamily?: string
  color?: string
  size?: number
}

const BaseText: React.FC<BaseTextProps> = ({
  text = '',
  fontFamily,
  color,
  size = 14,
  children,
  style = {},
  theme,
  ...rest
}) => {
  const styles = {
    fontFamily: fontFamily || theme?.fonts.rubikRegular,
    fontSize: size,
    color: color || theme?.colors.backBlack
  }

  return (
    <Text style={[styles, style]} {...rest}>
      {text}
      {children}
    </Text>
  )
}

export default withTheme<BaseTextProps>(BaseText)
