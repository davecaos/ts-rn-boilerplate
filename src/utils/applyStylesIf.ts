import {
  ViewStyle,
  FlexStyle,
  ImageStyle,
  TransformsStyle,
  TextStyle,
  ShadowStyleIOS,
  TextStyleAndroid,
  TextStyleIOS
} from 'react-native'

type applyStylesIfStylesParam =
  | ViewStyle
  | FlexStyle
  | ImageStyle
  | TransformsStyle
  | TextStyle
  | ShadowStyleIOS
  | TextStyleAndroid
  | TextStyleIOS

/**
 * @name applyStylesIf
 * @description If the condition provided in the first argument evaluates to true then apply the styles from the second argument, otherwise returns an empty object.
 */
export const applyStylesIf = (
  condition: boolean | undefined,
  styles: applyStylesIfStylesParam
): applyStylesIfStylesParam | Record<string, never> => (condition ? styles : {})
