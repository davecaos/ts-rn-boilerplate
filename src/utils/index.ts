export * from './switchCallback'
export * from './applyStylesIf'
