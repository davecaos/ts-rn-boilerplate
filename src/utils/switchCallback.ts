/**
 * @name switchCallback
 * @description Switch callback based on condition. If the condition evaluates to true, it returns the callback, otherwise returns an empty function.
 */
export const switchCallback = <C>(condition: boolean, callback: C): C | (() => void) => {
  /* eslint-disable-next-line @typescript-eslint/no-empty-function */
  if (!condition) return () => {}
  return callback
}
