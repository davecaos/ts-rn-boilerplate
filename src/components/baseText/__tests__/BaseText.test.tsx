import React from 'react'
import {render} from '../../../utils/test'
import BaseText from '../BaseText'

describe('BaseText', () => {
  test('Snapshot Test', () => {
    const {toJSON} = render(<BaseText text="Foo" />, {})
    expect(toJSON()).toMatchSnapshot()
  })
})
