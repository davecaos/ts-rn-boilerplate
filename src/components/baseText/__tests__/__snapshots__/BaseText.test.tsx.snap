// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseText Snapshot Test 1`] = `
<Text>
  Foo
</Text>
`;
